### ISSUE TITLE
Add the ability to edit the creative menu

### ISSUE THREAD

**Issue by toontownlpz** [CraftTweaker/CraftTweaker#1033]
**Feature request name**
Creative Menu Editor

**Feature request description**
I'd like to be able to do things like remove items from appearing in the creative menu, as well as remove tabs of my choice and move items between tabs to simply the presentation of the creative menu.

**Feature request reason**
I would love to be able edit my modpack in more detail to prevent the user obtaining certain items in creative mode. There are also a lot of items that have their own tabs that I don't really need on display, so removing them would clear up a lot of clutter. I would also like to be able to move items from a mod-specific tab to a Minecraft default tab like Combat or Foodstuffs.

**Comment by nepphhh** [comment]
It sounds like you want to perma-prune items/blocks from ever possibly existing in a world. Seems like forcing blocks & items out of memory could decrease the ram commitment as well... would be useful when trimming down big mods with things you don't want to include.

**Comment by toontownlpz** [comment]
Yeah that's essentially what I want to do. Would be very beneficial for my modpack.

**Comment by Ommina** [comment]
How is this going to reduce memory consumption?  The blocks and items are still going to be registered.  The textures will still be stitched.  /give will still function normally.  In what way does changing the list players _in creative_ see, beneficial?

**Comment by nepphhh** [comment]
@Ommina Let me rephrase it; _if_ you could find a way to allow CT to unregister the blocks & items and _not_ stitch the textures, then it would be beneficial.

Just hiding them from the creative menu would have some slight benefit inofitself, but that's solely out of concerns for limiting what the players of your modpack have access to--although, as you mention, /give would still work.

But of course, the ideal is a way to completely unload selected items & blocks from the game.

**Comment by kindlich** [comment]
I believe you have become a little side tracked but let me make this clear:
CraftTweaker cannot remove/"unregister" existing items from the game.

You asked for several possible features:

- Hiding an entire Creative Tab
- Removing an item from a creative tab
- Moving an item to a different creative tab


I don't think that's as easily doable, because every Item has methods to check if the item is inside a given Tab.
For hiding a tab we could maybe remove it from the list, but that could have unforeseen consequences from other mods.

I'll leave the issue open for now but honestly I don't think it's going to be implemented anytime soon.

**Comment by jaredlll08** [comment]
Honestly the more I think about it, I don't think touching tabs at all is a good idea. As kindlich said:
We (or any mod) can't / shouldn't remove items / blocks from the game, forge doesn't even want mods doing it with configs.

Everything else you've asked for is bound to fail since mods may be doing tricky things with their creative tabs and may not like their tab not existing at all.
